pkg/util/protocol/protocol.go

```diff
@@ -51,15 +51,14 @@ func getCommonProtocol(one, another core_mesh.Protocol) core_mesh.Protocol {
 	return core_mesh.ProtocolUnknown
 }
 
-// InferServiceProtocol returns a common protocol for a given group of endpoints.
 func InferServiceProtocol(endpoints []core_xds.Endpoint) core_mesh.Protocol {
```

Comment:
I think this should just move in to be a private function in mesh_context_builder to avoid further usage of this.